### PR TITLE
feat: support conditional axis + add examples

### PR DESCRIPTION
**Pull request by kanitw**
part of #2907

### CODE REVIEW

File: build/vega-lite-schema.json [hunk 1]
```diff
@@ -560,34 +560,76 @@
...
         "labelAlign": {
-          "$ref": "#/definitions/Align",
-          "description": "Horizontal text alignment of axis tick labels, overriding the default setting for the current axis orientation."
+          "anyOf": [
+            {
+              "$ref": "#/definitions/Align",
+              "description": "Horizontal text alignment of axis tick labels, overriding the default setting for the current axis orientation."
+            },
+            {
+              "$ref": "#/definitions/ConditionalAxisProperty<(def-alias-981995915-5990-6039-981995915-0-9900|undefined|null)>"
```
Comment (kanitw) [inline build/vega-lite-schema.json]: > def-alias-981995915-5990-6039-981995915-0-9900

@domoritz  I don't know why the JSON schema generator doesn't read the name `Align` from Vega and put it here. 

That said, the schema is still technically correct here.

Comment (domoritz) [reply build/vega-lite-schema.json]: I'd need to look more into this but it's not an easy type: `[k in ConditionalAxisProp]?: BaseAxisNoSignals[k] | ConditionalAxisProperty<BaseAxisNoSignals[k]>;`.

File: site/docs/encoding/axis.md [hunk 2]
```diff
@@ -101,6 +101,25 @@ For example, the following plot has a customized x-axis title.
 **TODO** (We have `encoding` property akin to [Vega's axis `encode`](https://vega.github.io/vega/docs/axes/#custom-axis-encodings), but within each element's block, we do not have `enter/update/exit`.)
 -->
 
+{:#conditional}
+
+### Conditional Axis Properties
+
+We can set axis properties (whose type include "ConditionalAxisProperty") to a [conditional value definition](condition.html#value).
```
Comment (domoritz) [inline site/docs/encoding/axis.md]:
```suggestion
We can set axis properties (which can be of type "ConditionalAxisProperty") to a [conditional value definition](condition.html#value).
```

File: src/axis.ts [hunk 3]
```diff
@@ -33,8 +35,119 @@ type BaseAxisNoSignals = AxisMixins &
...
+
+export type ConditionalAxisProperty<V extends Value | number[]> = ValueDef<V> & {
+  condition: ConditionalPredicate<ValueDef<V>> | ConditionalPredicate<ValueDef<V>>[];
+};
+
+export function isConditionalAxisValue<V extends Value | number[]>(v: any): v is ConditionalAxisProperty<V> {
+  return v['condition'];
+}
+
 // Vega axis config is the same as vega axis base. If this is not the case, add specific type.
```
Comment (domoritz) [inline src/axis.ts]:
```suggestion
// Vega axis config is the same as Vega axis base. If this is not the case, add specific type.
```

**Pull request merged**